packages/patternfly-4/react-core/src/components/Select/SelectToggle.js

```diff
@@ -0,0 +1,156 @@
+  /** Type of the toggle button, defaults to 'button' */
+  type: PropTypes.string,
+  /** Additional props are spread to the container <button> */
+  '': PropTypes.any
+};
+
+const defaultProps = {
+  children: null,
+  className: '',
+  isExpanded: false,
+  parentRef: null,
+  isFocused: false,
+  isHovered: false,
+  isActive: false,
+  isPlain: false,
+  onToggle: Function.prototype
+};
+
+class SelectToggle extends Component {
+  componentDidMount() {
+    document.addEventListener('mousedown', this.onDocClick);
+    document.addEventListener('touchstart', this.onDocClick);
+    document.addEventListener('keydown', this.onEscPress);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('mousedown', this.onDocClick);
+    document.removeEventListener('touchstart', this.onDocClick);
+    document.removeEventListener('keydown', this.onEscPress);
+  }
+
+  onDocClick = event => {
+    const { parentRef, isExpanded, onToggle, onClose } = this.props;
+    if (isExpanded && parentRef && !parentRef.contains(event.target)) {
+      onToggle && onToggle(false);
+      onClose && onClose();
+      this.toggle.focus();
+    }
+  };
+
+  onEscPress = event => {
+    const { parentRef, isExpanded, onToggle, onClose } = this.props;
+    if (!isExpanded && event.key === KeyTypes.Enter) this.toggle.focus();
+    if (
+      isExpanded &&
+      (event.key === KeyTypes.Escape || event.key === KeyTypes.Tab) &&
+      parentRef &&
+      parentRef.contains(event.target)
+    ) {
+      onToggle && onToggle(false);
+      onClose && onClose();
+      this.toggle.focus();
+    }
+  };
+
+  onKeyDown = event => {
+    const { isExpanded, onToggle, onClose, onEnter } = this.props;
+    if ((event.key === KeyTypes.Tab && !isExpanded) || (event.key !== KeyTypes.Enter && event.key !== KeyTypes.Space))
+      return;
+    event.preventDefault();
+    if ((event.key === KeyTypes.Tab || event.key === KeyTypes.Enter || event.key === KeyTypes.Space) && isExpanded) {
+      onToggle && onToggle(!isExpanded);
+      onClose && onClose();
+      this.toggle.focus();
+    } else if ((event.key === KeyTypes.Enter || event.key === KeyTypes.Space) && !isExpanded) {
+      onToggle(!isExpanded);
+      onEnter();
+    }
+  };
+
+  render() {
+    const {
+      className,
+      children,
+      isExpanded,
+      isFocused,
+      isActive,
+      isHovered,
+      isPlain,
+      onToggle,
+      onEnter,
+      onClose,
+      parentRef,
+      id,
+      type,
+      ...props
+    } = this.props;
+    return (
+      <button
+        {...props}
+        id={id}
+        ref={toggle => {
+          this.toggle = toggle;
+        }}
+        className={css(
+          styles.selectToggle,
+          isFocused && styles.modifiers.focus,
+          isHovered && styles.modifiers.hover,
+          isActive && styles.modifiers.active,
+          isPlain && styles.modifiers.plain,
+          className
+        )}
+        type={type || 'button'}
```

Comment:
Can just set 'button' as the default prop.

Reply:
Good catch, will clean up